File: java-plugin-netlogo-maven/src/main/java/org/test/BuildRandomList.java
package org.test;

import org.nlogo.api.*;
import java.util.Random;

public class BuildRandomList extends DefaultReporter {

    Random r = new Random();
    public Syntax getSyntax() {
        return Syntax.reporterSyntax(new int[]{Syntax.NumberType()},Syntax.ListType());
    }

    public Object report(Argument args[], Context context) throws ExtensionException, LogoException {
        LogoListBuilder list = new LogoListBuilder();

        int n;
        try {
            n = args[0].getIntValue();
        } catch (LogoException e) {
            throw new ExtensionException(e.getMessage());
        }

        for (int i = 0; i < n; i++) {
            list.add(Double.valueOf(r.nextDouble()));
        }
        // Java autobox double into Double
        return list;

    }

}

File: scala-plugin-netlogo-sbt/src/BuildRandomList.scala
package org.test

import org.nlogo.{ agent, api, nvm }
import java.util.Random
import api.Syntax._
import api.ScalaConversions._

class BuildRandomList extends api.DefaultReporter {
  val r = scala.util.Random

  override def getSyntax =
    reporterSyntax(Array(NumberType), ListType)
  def report(args: Array[api.Argument], context: api.Context): AnyRef = {
    val n = try args(0).getIntValue
    catch {
      case e: api.LogoException =>
        throw new api.ExtensionException(e.getMessage)
    }
    if (n < 0)
      throw new api.ExtensionException("input must be positive")
    (0 until n).map{_ => r.nextDouble}.toLogoList
  }
}